include/hipSYCL/sycl/handler.hpp

```diff
@@ -1012,6 +1013,8 @@ class handler {
   rt::runtime* _rt;
 
   bool _operation_uses_reductions = false;
+
+  std::shared_ptr<rt::kernel_cache> _kernel_cache;
```

Comment:
Rather than store an additional `shared_ptr` (which will need to be initialized for every kernel launch) could we not just store a copy of the `shared_ptr` inside the runtime, and then access that one in `handler`? After all, `handler` already stores a `runtime` pointer.